catalog/nsili/catalog-nsili-common/src/main/java/com/connexta/alliance/nsili/common/BqsConverter.java

```diff
@@ -0,0 +1,1160 @@
+                        currFilter = filterBuilder.not(filters.iterator()
+                                .next());
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void enterPrimary(BqsParser.PrimaryContext ctx) {
+            if (ctx.LIKE() != null) {
+                print("LIKE");
+                bqsOperatorStack.push(BqsOperator.LIKE);
+            } else if (ctx.EXISTS() != null) {
+                print("EXISTS");
+                bqsOperatorStack.push(BqsOperator.EXISTS);
+            } else if (ctx.OF() != null) {
+                print("OF");
+                bqsOperatorStack.push(BqsOperator.OF);
+            }
+        }
+
+        @Override
+        public void exitPrimary(BqsParser.PrimaryContext ctx) {
+            if (ctx.LIKE() != null) {
+                bqsOperatorStack.pop();
+            } else if (ctx.EXISTS() != null) {
+                bqsOperatorStack.pop();
+            } else if (ctx.OF() != null) {
+                bqsOperatorStack.pop();
+            }
+        }
+
+        @Override
+        public void enterAttribute_name(BqsParser.Attribute_nameContext ctx) {
+            print(ctx.getText());
+        }
+
+        @Override
+        public void exitAttribute_name(BqsParser.Attribute_nameContext ctx) {
+            attribute = NsiliAttributeMap.getDdfAttributeForNsili(ctx.getText());
+        }
+
+        @Override
+        public void enterComp_op(BqsParser.Comp_opContext ctx) {
+            print(ctx.getText());
+            if (ctx.EQUAL() != null) {
+                bqsOperatorStack.push(BqsOperator.EQUAL);
+            } else if (ctx.NOTEQ() != null) {
+                bqsOperatorStack.push(BqsOperator.NOT);
+            } else if (ctx.GT() != null) {
+                bqsOperatorStack.push(BqsOperator.GT);
+            } else if (ctx.GTE() != null) {
+                bqsOperatorStack.push(BqsOperator.GTE);
+            } else if (ctx.LT() != null) {
+                bqsOperatorStack.push(BqsOperator.LT);
+            } else if (ctx.LTE() != null) {
+                bqsOperatorStack.push(BqsOperator.LTE);
+            }
+        }
+
+        @Override
+        public void exitComp_op(BqsParser.Comp_opContext ctx) {
+
+        }
+
+        @Override
+        public void enterConstant_expression(BqsParser.Constant_expressionContext ctx) {
+
+        }
+
+        @Override
+        public void exitConstant_expression(BqsParser.Constant_expressionContext ctx) {
+            BqsOperator bqsOperator = bqsOperatorStack.pop();
+            boolean shouldNegate = false;
+            if (bqsOperatorStack.peek() == BqsOperator.NOT) {
+                shouldNegate = true;
+            }
+
+            if (!dateStr.isEmpty()) {
+                Filter filter = null;
+                try {
+                    Date date;
+                    try {
+                        SimpleDateFormat dateFormat = new SimpleDateFormat(BQS_FULL_DATE_FORMAT);
```

Comment:
You can use joda-time to make static thread safe formatters so you don't have to make new instances here each time.